### ISSUE TITLE
Implement JPEG decoding via nvjpeg

### ISSUE THREAD

**Issue by fmassa** [pytorch/vision#2742]
## 🚀 Feature

We currently provide CPU image decoding functions for PNG and JPEG. It could be interesting to consider using nvjpeg to perform decoding directly on the GPU, as it could benefit certain workloads.

## Motivation

With CUDA 10.0, nvjpeg is [part of the CUDA toolkit](https://developer.nvidia.com/nvjpeg), which means that it would be easily accessible for us to use it.

The API for the decoding function is not yet clear to me, and would require some digging
Ideally, passing a CUDA tensor for `decode_jpeg` should do the decoding on the GPU, although I'm not sure if that's how nvjpeg works internally.

**Comment by jamt9000** [comment]
It looks like it requires a host pointer and does some preprocessing on the cpu


> | Parameter |  Input / Output |  Memory  | Description |
> |-------------------------|---|---|----|
> | const unsigned char *data |  Input | Host  | Pointer to the encoded data.
> | nvjpegImage_t *destination |  Input/Output | Host/ Device | Pointer to the structure that describes the output destination. This structure should be on the host (CPU), but the pointers in this structure should be pointing to the device (i.e., GPU) memory. See nvjpegImage_t.

so passing gpu data directly wouldn't work

**Comment by fmassa** [comment]
Yes, that was my understanding as well. We would need to see what potential speed benefits single-image decoding could bring compared to say turbo-jpeg, and if the speed improvements are there what would be a good API for using it. Passing a `device` flag to `decode_jpeg`?
